[Request] Use dependency versions matching the peer dependencies of consumed libraries

#### Is this a bug, enhancement, or feature request?
bug/feature request

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
```
"@angular/core": "~15.2.2",
"@fundamental-ngx/core": "0.40.0-rc.12",
"@fundamental-ngx/i18n": "0.40.0-rc.12",
"@fundamental-ngx/platform": "0.40.0-rc.12",
"fundamental-styles": "0.27.1",
"@sap-theming/theming-base-content": "11.2.1",
```

#### Briefly describe your proposal.
**Take care that dependency versions and peer dependencies are matching.**
Using the versions above, we run into the following issue:
<details>
  <summary>npm install (output)</summary>
  
  ```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: fundamental-styles@0.27.1
npm ERR! Found: @sap-theming/theming-base-content@11.2.1
npm ERR! node_modules/@sap-theming/theming-base-content
npm ERR!   @sap-theming/theming-base-content@"11.2.1" from the root project
npm ERR!   @sap-theming/theming-base-content@"^11.2.0" from @fundamental-ngx/cdk@0.40.0-rc.12
npm ERR!   node_modules/@fundamental-ngx/cdk
npm ERR!     @fundamental-ngx/cdk@"0.40.0-rc.12" from @fundamental-ngx/core@0.40.0-rc.12
npm ERR!     node_modules/@fundamental-ngx/core
npm ERR!       @fundamental-ngx/core@"0.40.0-rc.12" from the root project
npm ERR!       1 more (@fundamental-ngx/platform)
npm ERR!     @fundamental-ngx/cdk@"0.40.0-rc.12" from @fundamental-ngx/platform@0.40.0-rc.12
npm ERR!     node_modules/@fundamental-ngx/platform
npm ERR!       @fundamental-ngx/platform@"0.40.0-rc.12" from the root project
npm ERR!   1 more (@fundamental-ngx/core)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @sap-theming/theming-base-content@"11.1.49" from fundamental-styles@0.27.1
npm ERR! node_modules/fundamental-styles
npm ERR!   fundamental-styles@"0.27.1" from the root project
npm ERR!   fundamental-styles@"0.27.1" from @fundamental-ngx/cdk@0.40.0-rc.12
npm ERR!   node_modules/@fundamental-ngx/cdk
npm ERR!     @fundamental-ngx/cdk@"0.40.0-rc.12" from @fundamental-ngx/core@0.40.0-rc.12
npm ERR!     node_modules/@fundamental-ngx/core
npm ERR!       @fundamental-ngx/core@"0.40.0-rc.12" from the root project
npm ERR!       1 more (@fundamental-ngx/platform)
npm ERR!     1 more (@fundamental-ngx/platform)
npm ERR!   1 more (@fundamental-ngx/core)
npm ERR! 
npm ERR! Conflicting peer dependency: @sap-theming/theming-base-content@11.1.49
npm ERR! node_modules/@sap-theming/theming-base-content
npm ERR!   peer @sap-theming/theming-base-content@"11.1.49" from fundamental-styles@0.27.1
npm ERR!   node_modules/fundamental-styles
npm ERR!     fundamental-styles@"0.27.1" from the root project
npm ERR!     fundamental-styles@"0.27.1" from @fundamental-ngx/cdk@0.40.0-rc.12
npm ERR!     node_modules/@fundamental-ngx/cdk
npm ERR!       @fundamental-ngx/cdk@"0.40.0-rc.12" from @fundamental-ngx/core@0.40.0-rc.12
npm ERR!       node_modules/@fundamental-ngx/core
npm ERR!         @fundamental-ngx/core@"0.40.0-rc.12" from the root project
npm ERR!         1 more (@fundamental-ngx/platform)
npm ERR!       1 more (@fundamental-ngx/platform)
npm ERR!     1 more (@fundamental-ngx/core)
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
</details>

It would be more consumable if fundamental-ngx would consume the same @sap-theming version as fundamental-styles. We can configure npm install and ci to resolve it using the legacy-peer-deps flag, but [npm maintainer recommend fixing the dependency graph](https://github.com/npm/cli/issues/4998) which we can't do in our project if the dependency graph is broken in a lib that we consume.

#### If this is a bug, please provide steps for reproducing it.
Use the versions above and run npm install. Make sure to disable the flags force and legacy-peer-deps.

#### Please provide relevant source code if applicable.

#### Is there anything else we should know?
Please be welcome to reach out, also using SAP internal channels. I'm open to test the next release candidate with our project before your actual release 🙂

--- a/package.json
+++ b/package.json
@@ -62,7 +62,7 @@
     "fast-deep-equal": "3.1.3",
     "focus-trap": "7.1.0",
     "focus-visible": "5.2.0",
-    "fundamental-styles": "0.28.2",
+    "fundamental-styles": "0.28.3",
     "highlight.js": "11.7.0",
     "intl": "1.2.5",
     "karma-viewport": "1.0.8",

